server/server.go

```diff
@@ -97,12 +97,20 @@ type GCAServer struct {
 	skipInvariants bool           // If set to true, 'CheckInvariants()' will not run on Close()
 	udpPort        uint16         // The port that the UDP conn is listening on
 	tcpPort        uint16         // The port that the TCP listener is using
+	allowIntApis   bool           // Enables bench testing the server with production settings
 	mu             sync.Mutex
 	tg             threadgroup.ThreadGroup
 
 	ApiArchiveRateLimiter *glow.RateLimiter // Rate limiter for the /archive endpoint.
 }
 
+// Enable bench test mode. Allows internal APIs and sets the server log level higher.
+func (gcas *GCAServer) EnableBenchTestMode() {
+	gcas.allowIntApis = true
+	gcas.logger.level = INFO
+	gcas.logger.Info("Bench test mode enabled")
```

Comment:
setting the log level should be done when the gcas is created, otherwise we're going to lose track of when and how the log level is being set.

Reply:
specifically when the log is created in the New() function of the gcas. If it gets changed anywhere else, its going to be confusing

Reply:
I put all the initialization steps in New() function and removed the "magic" api. It is much easier to follow the initialization now.